### PR TITLE
Liuliu/mobile database page test

### PR DESCRIPTION
**Pull request by liuliu-dev**
sorry that I was not able to split them up. The major changes are testRepoPageMobile.ts and repoHeaderNav.ts, used for testing the header and content on mobile pages. Then added that mobile tests to these files:
publicPaths>render:
>commits
>databases>all files under this folder
>issues
>pulls
>webhooks
>workspaces

a fix while making those tests:
1. the viewport was reset for every test that is not using the default viewport, this makes the tests slow and flaky when the viewport was kept switching for the tests. So I moved the viewport dimension outside, this makes the tests run much faster.

`
describe.skip(
        d.description,
        {
          viewportHeight: deviceDimensions[d.device].height,
          viewportWidth: deviceDimensions[d.device].width,
        },
        () => {
          runTests({ ...d, currentPage });
        },
      );
`

### CODE REVIEW

**Review by cheshirekate8 — APPROVED** [review]
Good job!! :D

File: cypress/integration/utils/index.ts [hunk 1]
```diff
@@ -99,15 +118,28 @@ export function runTestsForDevices({
...
     if (skip || skipForLogin) {
-      describe.skip(d.description, () => {
-        runTests({ ...d, currentPage });
-      });
+      describe.skip(
+        d.description,
+        {
+          viewportHeight: deviceDimensions[d.device].height,
+          viewportWidth: deviceDimensions[d.device].width,
+        },
```
Comment (tbantle22) [inline cypress/integration/utils/index.ts:127]: If you made the type of `deviceDimensions` `Record<Cypress.ViewportPresent, Cypress.Viewport>` you could just do this here
```suggestion
       deviceDimensions[d.device],
```

**Review by tbantle22 — APPROVED** [review]
A few suggestions and be careful of the imports that aren't relative paths but otherwise looks great!

File: cypress/integration/utils/index.ts [hunk 2]
```diff
@@ -40,13 +60,12 @@ export function runTests({
 }: TestsArgs) {
   before(() => {
     cy.visitPage(currentPage, loggedIn);
+    cy.visitViewport(device);
```
[the remark this answers is not in the document]
Comment (liuliu-dev) [reply cypress/integration/utils/index.ts:66]: yeah, I am quite excited about this change! can write a blog about the things I learned through building and testing the mobile pages after the pages are done.

File: cypress/integration/productionTests/publicPaths/render/database/ngrams.spec.ts [hunk 3]
```diff
@@ -15,6 +12,14 @@ import {
...
 import { testRepoHeaderWithBranch } from "../../../../utils/sharedTests/repoHeaderNav";
+import {
+  tableExpectations,
+  testQueryCatalogSection,
+  testSchemaSection,
+  testViewsSection,
+} from "../../../../utils/sharedTests/repoLeftNav";
+import { testSqlConsole } from "../../../../utils/sharedTests/sqlEditor";
+import { mobileTests } from "../../../../utils/sharedTests/testRepoPageMobile";
 // import { testSqlConsole } from "../../../../utils/sharedTests/sqlEditor";
```
Comment (tbantle22) [inline cypress/integration/productionTests/publicPaths/render/database/ngrams.spec.ts:23]: Can this be deleted?

File: cypress/integration/utils/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (tbantle22) [reply cypress/integration/utils/index.ts:66]: That would be great!

**Review by tbantle22 — APPROVED** [review]
Awesome, good to merge!

**Pull request merged**